### DISCUSSION TITLE
Persistence of app.storage.general: What is the median lifetime

### DISCUSSION THREAD

**Discussion by vickorian** [zauberzeug/nicegui#2972] [Q&A] [answered]
### Question

I'm wondering what the lifetime is for app.storage.general. I'd like to avoid having to call a section of code every page as all users will be accessing the shared database.

Any help would be greatly appreciated.

**Comment by falkoschindler** [comment, marked as answer]
Hi @vickorian,

Currently the storage isn't pruned at all. There's an open feature request for auto-cleaning the stored JSON files: #991.

> I'd like to avoid having to call a section of code every page as all users will be accessing the shared database.

Sorry, I don't understand what you're saying.

**Reply by vickorian** [reply]
@falkoschindler So if the storage has infinite lifetime then i can make it available in the app itself and have the app handle storage and retrieval instead of creating and recalling variables every time. But if that storage is going to have a finite lifetime and be pruned, then that wouldn't be a good option. Basically i needed variables to be available to all users.
